core/src/main/scala/cats/mtl/MonadChronicle.scala

```diff
@@ -0,0 +1,29 @@
+package cats
+package mtl
+
+import cats.data.Ior
+
+trait MonadChronicle[F[_], E] extends Serializable {
+  val monad: Monad[F]
+
+  def dictate(c: E): F[Unit]
+
+  def disclose[A](c: E)(implicit M: Monoid[A]): F[A]
+
+  def confess[A](c: E): F[A]
+
+  def memento[A](ma: F[A]): F[Either[E, A]]
```

Comment:
After looking at this for a long time, I think we could radically make this type class simpler by returning `F[Ior[E, A]]` here instead. WDYT?

Reply:
/cc @SystemFw @andyscott

Reply:
I'm going to go ahead and implement this and see how it plays out with deriving the other combinators while we wait for more input :)